autonomy/cli/scaffold_fsm.py

```diff
@@ -495,6 +496,96 @@ def _get_round_behaviour_section(self) -> str:
         )
 
 
+class PayloadsFileGenerator(AbstractFileGenerator):
+    """File generator for 'payloads.py' modules."""
+
+    FILENAME = PAYLOADS_FILENAME
+
+    PAYLOADS_FILE = dedent(
+        """\
+        \"\"\"This module contains the transaction payloads of the {FSMName}.\"\"\"
+
+        from enum import Enum
+        from typing import Any, Dict, Hashable, Optional
+
+        from packages.valory.skills.abstract_round_abci.base import BaseTxPayload
+
+
+        class TransactionType(Enum):
+            \"\"\"Enumeration of transaction types.\"\"\"
+
+            # TODO: define transaction types: e.g. TX_HASH: "tx_hash"
+            ...
+
+            def __str__(self) -> str:
+                \"\"\"Get the string value of the transaction type.\"\"\"
+                return self.value
+
+        """
+    )
+
+    BASE_PAYLOAD_CLS = dedent(
+        """\
+        class Base{FSMName}Payload(BaseTxPayload):
+            \"\"\"Base payload for {FSMName}.\"\"\"
+
+            def __init__(self, sender: str, content: Hashable, **kwargs: Any) -> None:
+                \"\"\"Initialize a 'select_keeper' transaction payload.\"\"\"
+
+                super().__init__(sender, **kwargs)
+                setattr(self, f"_{{self.transaction_type}}", content)
+                p = property(lambda s: getattr(self, f"_{{self.transaction_type}}"))
+                setattr(self.__class__, f"{{self.transaction_type}}", p)
+
+            @property
+            def data(self) -> Dict[str, Hashable]:
+                \"\"\"Get the data.\"\"\"
+                return {{str(self.transaction_type): getattr(self, str(self.transaction_type))}}
+
+    """
+    )
```

Comment:
I introduced this because it's much easier to scaffold the Payload classes, which now only need to set the `transaction_type`